src/main/java/io/github/battlepass/quests/service/executor/QuestExecutionBuilder.java

```diff
@@ -0,0 +1,82 @@
+package io.github.battlepass.quests.service.executor;
+
+import io.github.battlepass.objects.quests.variable.QuestResult;
+import io.github.battlepass.quests.service.base.ExternalQuestContainer;
+import io.github.battlepass.quests.service.base.QuestContainer;
+import io.github.battlepass.service.CheckHelper;
+import me.hyfe.simplespigot.annotations.NotNull;
+import org.bukkit.block.Block;
+import org.bukkit.entity.Player;
+import org.bukkit.inventory.ItemStack;
+
+import javax.annotation.CheckReturnValue;
+import java.math.BigInteger;
+import java.util.function.UnaryOperator;
+
+public interface QuestExecutionBuilder {
+
+    void buildAndExecute() throws IllegalStateException;
+
+    @NotNull
+    QuestExecution build() throws IllegalStateException;
+
+    @CheckReturnValue
+    @NotNull
+    static QuestExecutionBuilder of(@NotNull QuestContainer container, @NotNull String questType) {
+        String buildType;
+        if (container instanceof ExternalQuestContainer) {
+            buildType = ((ExternalQuestContainer) container).getPrefix().concat(questType);
+        } else {
+            buildType = questType;
+        }
+        return new QuestExecutionBuilderImpl(container.getQuestPipeline(), buildType);
+    }
```

Comment:
Maybe move this method to the top or bottom of the interface?

Reply:
Moved to top